server/src/main/java/org/elasticsearch/discovery/SeedHostsResolver.java

```diff
@@ -58,8 +74,93 @@ public SeedHostsResolver(String nodeName, Settings settings, TransportService tr
         this.nodeName = nodeName;
         this.transportService = transportService;
         this.hostsProvider = seedProvider;
-        resolveTimeout = UnicastZenPing.getResolveTimeout(settings);
-        concurrentConnects = UnicastZenPing.getMaxConcurrentResolvers(settings);
+        resolveTimeout = getResolveTimeout(settings);
+        concurrentConnects = getMaxConcurrentResolvers(settings);
+    }
+
+    public static int getMaxConcurrentResolvers(Settings settings) {
+        return DISCOVERY_SEED_RESOLVER_MAX_CONCURRENT_RESOLVERS_SETTING.get(settings);
+    }
+
+    public static TimeValue getResolveTimeout(Settings settings) {
+        return DISCOVERY_SEED_RESOLVER_TIMEOUT_SETTING.get(settings);
+    }
+
+    /**
+     * Resolves a list of hosts to a list of transport addresses. Each host is resolved into a transport address (or a collection of
+     * addresses if the number of ports is greater than one). Host lookups are done in parallel using specified executor service up
+     * to the specified resolve timeout.
+     *
+     * @param executorService  the executor service used to parallelize hostname lookups
+     * @param logger           logger used for logging messages regarding hostname lookups
+     * @param hosts            the hosts to resolve
+     * @param limitPortCounts  the number of ports to resolve (should be 1 for non-local transport)
+     * @param transportService the transport service
+     * @param resolveTimeout   the timeout before returning from hostname lookups
+     * @return a list of resolved transport addresses
+     */
+    public static List<TransportAddress> resolveHostsLists(
```

Comment:
Could we make this into an instance method rather than a static method that accepts half of the fields of this class as arguments?

Reply:
see 0bf8a12